docker/pw_discord_bot/Dockerfile

```diff
@@ -1,15 +1,21 @@
-FROM golang:1.17-alpine AS build
+FROM golang:1.18-alpine AS build
 
 # Set necessary environmet variables needed for our image
 ENV GO111MODULE=on \
     CGO_ENABLED=0 \
     GOOS=linux \
-    GOARCH=amd64
+    GOARCH=amd64 \
+    GOAMD64=v3
 
 WORKDIR /app
 
+# prefill the build steps
+COPY go.mod /app
+COPY go.sum /app
```

Comment:
This is a minor nit-pick but this actually creates two Docker layers and increases the image size, you can do the copy in one:
```Dockerfile
COPY go.mod go.sum /app
```

Reply:
these layers does not appear in the final image though.

The only things in the final docker image are the built binary and Alpine